vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/AbstractConfigurationObject.java

```diff
@@ -45,6 +51,29 @@ public void setId(String id) {
 
     public abstract String getType();
 
+    protected void addChild(AbstractConfigurationObject configurationObject) {
+        children.add(configurationObject);
```

Comment:
In order to retrieve changes from the whole hierarchy, we also need to keep track of each reference that is used in an object. Maybe addReference / removeReference might be a better name.